client/src/App.tsx

```
@@ -1,35 +1,119 @@
-import reactLogo from './assets/react.svg';
-import viteLogo from '/vite.svg';
 import './App.css';
+import { Routes, Route } from 'react-router-dom';
+import Header from './components/Header';
+import Ideas from './pages/Ideas';
+import MyRecipes from './pages/MyRecipes';
+import Favorites from './pages/Favorites';
+import SignInForm from './pages/SignInForm';
+import SignUpForm from './pages/SignUpForm';
+import Details from './pages/Details';
+import RecipeForm from './pages/RecipeForm';
+import { User, UserProvider } from './components/UserContext';
+import { insertComment, readUser, saveToken, saveUser } from './lib/data';
+import { useEffect, useState } from 'react';
+import SearchList from './pages/SearchList';
 
 export default function App() {
-  const [serverData, setServerData] = useState('');
-
+  const [user, setUser] = useState<User>();
+  const [token, setToken] = useState<string>();
+  const [error, setError] = useState<unknown>();
   useEffect(() => {
-    async function readServerData() {
-      const resp = await fetch('/api/hello');
-      const data = await resp.json();
+    const userSession = readUser();
+    if (userSession !== null) {
+      setUser(userSession);
+    }
+  }, []);
+
+  function handleSignIn(user: User, token: string) {
+    setUser(user);
+    setToken(token);
+    saveToken(token);
+    saveUser(user);
+  }
 
-      console.log('Data from server:', data);
+  function handleSignOut() {
+    setUser(undefined);
+    setToken(undefined);
+    saveToken(undefined);
+    saveUser(undefined);
+  }
 
-      setServerData(data.message);
+  async function handleCommentPost(message: string, recipeId: string) {
+    if (user) {
+      const messageObject = {
+        userId: user.userId,
+        message,
+        recipeId,
+      };
+      try {
+        await insertComment(messageObject);
+        alert('message posted.');
+      } catch (error) {
+        console.error(error);
+        setError(error);
+      }
+    } else {
```

I prefer to "fail fast" and put this check and alert at the top of the function (remembering to return after the alert).